sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/query/DocumentQueryExecutionContextFactory.java

```diff
@@ -92,6 +93,22 @@
 
         Instant startTime = Instant.now();
         Mono<PartitionedQueryExecutionInfo> queryExecutionInfoMono;
+
+        if (ImplementationBridgeHelpers
+            .CosmosQueryRequestOptionsHelper
+            .getCosmosQueryRequestOptionsAccessor()
+            .isQueryPlanRetrievalDisallowed(cosmosQueryRequestOptions)) {
+
+            Instant endTime = Instant.now(); // endTime for query plan diagnostics
+
+            return getTargetRangesFromEmptyQueryPlan(
```

Comment:
Do we need this additional function?
`getTargetRangesFromQueryPlan` would essentially skip the queryRanges from partitionedQueryExecutionInfo when feedRange  is provided in the queryRequestoptions which achieves same behaviour as the new method `getTargetRangesFromEmptyQueryPlan`. You may just have to add some null checks in `getTargetRangesFromQueryPlan` as required

Reply:
True - but isn't it easeir to read this way? - having a method called GetxxformQueryPlan where Query Plan can be null/empty is weird.